net/yggdrasil/files/ygguci

```diff
@@ -0,0 +1,247 @@
+      table.insert(obj.SessionFirewall.BlacklistEncryptionPublicKeys, s.key)
+  end)
+  -- /session firewall config
+
+  -- tunnel routing config
+  if config.TunnelRouting_Enable ~= nil then
+      obj.TunnelRouting.Enable = to_bool(config.TunnelRouting_Enable)
+  end
+  cursor:foreach("yggdrasil", "ipv6_remote_subnet", function (s) 
+      obj.TunnelRouting.IPv6RemoteSubnets[s.subnet] = s.key
+  end)
+  cursor:foreach("yggdrasil", "ipv6_local_subnet", function (s) 
+      table.insert(obj.TunnelRouting.IPv6LocalSubnets, s.subnet)
+  end)
+  cursor:foreach("yggdrasil", "ipv4_remote_subnet", function (s) 
+      obj.TunnelRouting.IPv4RemoteSubnets[s.subnet] = s.key
+  end)
+  cursor:foreach("yggdrasil", "ipv4_local_subnet", function (s) 
+      table.insert(obj.TunnelRouting.IPv4LocalSubnets, s.subnet)
+  end)
+  -- /tunnel routing config
+
+  if config.SwitchOptions_MaxTotalQueueSize ~= nil then
+      obj.SwitchOptions.MaxTotalQueueSize = tonumber(config.SwitchOptions_MaxTotalQueueSize)
+  end
+
+  return obj
+end
+
+--- Parse and save updated configuration from JSON input
+--
+-- Transforms general settings into UCI sections, and replaces the UCI config's
+-- contents with them.
+-- @param table JSON input
+-- @return Boolean whether saving succeeded
+function UCI.set(obj)
+  local cursor = uci.cursor()
+
+  for i, section in pairs(cursor:get_all("yggdrasil")) do
+    cursor:delete("yggdrasil", section[".name"])
+  end
+
+
+  cursor:set("yggdrasil", "yggdrasil", "yggdrasil")
+  cursor:set("yggdrasil", "yggdrasil", "EncryptionPublicKey", obj.EncryptionPublicKey) 
+  cursor:set("yggdrasil", "yggdrasil", "EncryptionPrivateKey", obj.EncryptionPrivateKey) 
+  cursor:set("yggdrasil", "yggdrasil", "SigningPublicKey", obj.SigningPublicKey) 
+  cursor:set("yggdrasil", "yggdrasil", "SigningPrivateKey", obj.SigningPrivateKey)
+  cursor:set("yggdrasil", "yggdrasil", "AdminListen", obj.AdminListen) 
+  cursor:set("yggdrasil", "yggdrasil", "IfName", obj.IfName) 
+  cursor:set("yggdrasil", "yggdrasil", "NodeInfoPrivacy", to_int(obj.NodeInfoPrivacy)) 
+  cursor:set("yggdrasil", "yggdrasil", "IfTAPMode", to_int(obj.IfTAPMode)) 
+  cursor:set("yggdrasil", "yggdrasil", "LinkLocalTCPPort", obj.LinkLocalTCPPort)
+  cursor:set("yggdrasil", "yggdrasil", "IfMTU", obj.IfMTU)
+
+  set_values(cursor, "peer", "uri", obj.Peers)
+  set_values(cursor, "listen_address", "uri", obj.Listen)
+  set_values(cursor, "multicast_interface", "name", obj.MulticastInterfaces)
+  set_values(cursor, "allowed_encryption_public_key", "key", obj.AllowedEncryptionPublicKeys)
+
+  for interface, peers in pairs(obj.InterfacePeers) do
+      for _, v in pairs(peers) do
+          local name = cursor:add("yggdrasil", "interface_peer")
+          cursor:set("yggdrasil", name, "interface", interface)
+          cursor:set("yggdrasil", name, "uri", v)
+      end
+  end
+
+  cursor:set("yggdrasil", "NodeInfo", "yggdrasil")
```

Comment:
Can we please lowercase all UCI section names? As far as I see this is the only occurrence for sections.

Reply:
Done